dashboard/app/controllers/api/v1/assessments_controller.rb

```diff
@@ -79,135 +79,168 @@ def index
   def section_responses
     responses_by_student = {}
 
+    script_id = @script.id
     assessment_script_levels = @script.get_assessment_script_levels
 
-    @section.students.each do |student|
-      # Initialize student hash
-      student_hash = {
-        student_name: student.name
-      }
-      responses_by_level_group = {}
-
+    @section.students.in_batches(of: 20).each do |student_batch|
```

Comment:
Wow, there's a lot going on here, thanks for taking on this monster! Is there a way it could be refactored into smaller methods to make it more readable/testable?

Reply:
Good question!  I actually started down this path but was worried about the risk.  Right now, if you do the compare with "ignore whitespace", you should be able to see exactly now the original logic of parsing the json was replaced with getting the answers from the individual questions (with the rest of the logic essentially untouched).

I'll take another pass and see it looks like.  In the meantime, does the approach seem generally reasonable?

Reply:
The approach seems reasonable to me, I just wasn't sure I'd be able to leave a quality review with all that's going on! My only concern would be that someone would try to reference the old JSON blob between now and when the clean-up happens (especially if it gets delayed more than a year).

Reply:
I did a bit of refactoring, see if this is a little easier to understand!  I think a good place to start is to read the header comment for `section_responses` to understand what the output of this function is supposed to look like.

Reply:
Thanks for breaking this up!!